BIDLicenses.js

```diff
@@ -38,33 +38,46 @@ const getCurrentLicense = async (licenseKey, serviceUrl, myKeyPair, requestUID =
         return infraKey;
     }
 
-    let cacheKey = `${serviceUrl}/${licenseKey}`;
-
-    let pubicKeyUrl = `${serviceUrl}/publickeys`;
-    let publicKey = (await WTM.executeRequest({
-        method: 'get',
-        url: pubicKeyUrl,
-        Logger,
-        requestUID,
-        cacheKey: pubicKeyUrl,
-        ttl: 600
-    })).json.publicKey;
-
-    let sharedKey = BIDECDSA.createSharedKey(myKeyPair.keySecret, publicKey);
-
     const requestId = JSON.stringify({
         ts: Math.round(new Date().getTime() / 1000),
         appid: senderId,
         uuid: requestUID
     });
 
-    const headers = {
-        licensekey: BIDECDSA.encrypt(licenseKey, sharedKey),
-        requestid: BIDECDSA.encrypt(requestId, sharedKey),
-        publickey: myKeyPair.keyId
-    };
+    let headers;
+    let u1Route = false;
+    if (!myKeyPair) {
+        u1Route = true;
+        headers = {
+            licensekey: licenseKey,
+            requestid: requestId
+        };
 
-    let url = `${serviceUrl}/servicekey/current`;
+    }
+
+    let cacheKey = `${serviceUrl}/${licenseKey}`;
+
+    if (!u1Route) { 
```

Comment:
@bill8575 if you want to keep U1 as well as ecdsa routes alive in this integration, you should consider having two different functions instead of having `if/else` blocks